src/utils.ts

```diff
@@ -513,28 +521,33 @@ export namespace Utils {
    */
   export function gatherTxSigners(
     transaction: Transaction,
-    accountIDs: string[],
+    signers: string[],
   ): string[] {
     const hashedSignatureBase = transaction.hash();
 
     const txSignatures = clone(transaction.signatures);
     const signersFound = new Set<string>();
 
-    for (const signer of accountIDs) {
+    for (const signer of signers) {
       if (txSignatures.length === 0) {
         break;
       }
 
-      const keypair = Keypair.fromPublicKey(signer);
+      // Raise an error on invalid G signers:
+      if (!StrKey.isValidEd25519PublicKey(signer)) {
+        throw new InvalidSep10ChallengeError("Signer is not a valid address");
+      }
+
+      const keypair = Keypair.fromPublicKey(signer); // This can throw new Error('Invalid Stellar public key');
```

Comment:
You don't need to call `isValidEd25519PublicKey` before calling `fromPublicKey`. The latter will raise an error if it is not valid.